package.json

```diff
@@ -43,7 +43,15 @@
       "/node_modules/",
       "/test/utils/",
       "/test/definitions/"
-    ]
+    ],
+    "coverageThreshold": {
+      "global": {
+        "branches": 100,
+        "functions": 100,
+        "lines": 100,
+        "statements": 100
```

Comment:
![](http://www.quickmeme.com/img/66/668f2c316d8211506929d9c3d3a2a2f3d65f130d512411d186520e34d559dce7.jpg)

😆 

Jokes aside, this sounds like a good idea (every commit passes!), but it can shut down contributions before they're even _committed_ because it won't be possible to commit (properly) without 100% test coverage.

This might make contributing to the project much harder, especially for people with less experience with tests. Even if we won't merge a PR without 100% coverage, the first _n_ commits might have lower coverage and we should still encourage the PR even if they need help.

What you do you think?

Reply:
Good point, but this doesn't actually prevent you from committing if the test coverage is < 100% - it just gives you a warning about that when you directly run `yarn test`

Reply:
Are you sure? (Not doubting you know, just want to be sure we know.)

The docs say it will make `jest` fail ([here](https://facebook.github.io/jest/docs/en/configuration.html#coveragethreshold-object)) and we're running `jest` in the `lint-staged` process. Would that command fail and block the commit?

I don't mind if it considers the tests a fail for `yarn test`, but I don't want to block commits.

Reply:
no, I tested when making this change and it doesn't block the commit

Reply:
Ok, then ignore the above. Thanks for the info!

Reply:
it **does** prevent commits when the lint-staged config uses "`test`" instead of `"jest"`

```js
"lint-staged": {
  "*.js": [
    "prettier-standard",
    "git add",
    "standard",
    // "jest --bail --findRelatedTests"
    "test --bail --findRelatedTests"
  ]
}
```

Reply:
I did some test locally with this branch and as far as I can tell it’s only when the `—coverage` flag is used that it will exit as a failure. We’re using the flag in the `test` script but not in the `lint-staged` script.

The jest docs are a little ambiguous here (it says it will fail jest, but doesn’t say you _must_ use the coverage flag).

Still reserving the right to be wrong.

Reply:
the issue with lint-staged is that it will only run for the changed files that lint-staged tells it to run. this makes the process faster because it does not run all tests. --coverage flag does not apply in this case. 

I think the setup @raingerber made is just enough to ensure testing passes when ran directly, and also at CI level through travis it will prevent going to coveralls if jest fails. so we have faster PR checks.